hack/shellcheck.sh

```diff
@@ -1,19 +1,33 @@
-#!/bin/sh
-
-set -eux
-
-if [ "${IS_CONTAINER:-}" != "" ]; then
-  TOP_DIR="${1:-.}"
-  find "${TOP_DIR}" \
-    -path "${TOP_DIR}/vendor" -prune \
-    -o -path "${TOP_DIR}/jsonnet/vendor" -prune \
-    -o -type f -name '*.sh' -exec shellcheck --format=gcc {} \+
-else
-  docker run --rm \
-    --env IS_CONTAINER=TRUE \
-    --volume "${PWD}:/workdir:ro,z" \
-    --entrypoint sh \
-    --workdir /workdir \
-    koalaman/shellcheck-alpine:stable \
-    /workdir/hack/shellcheck.sh "${@}"
-fi;
+#!/bin/bash
+
+set -euo pipefail
+
+TMP_BIN="$(pwd)/tmp/bin"
+
+# Ensure that we use the binaries from the versions defined in hack/tools/go.mod.
+PATH="${TMP_BIN}:${PATH}"
+
+install() {
+	version="stable" # or "v0.7.2", or "latest"
+	platform=""
+	if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+		platform="linux"
+	elif [[ "$OSTYPE" == "darwin"* ]]; then
+		platform="darwin"
+	fi
+
+	mkdir -p "${TMP_BIN}"
+	cd "${TMP_BIN}" || exit 1
```

Comment:
we don't need to check for errors here since `-e` is set.

Reply:
shellcheck wanted it :P